### DISCUSSION TITLE
Why I can't name nested field arrays within field arrays?

### DISCUSSION THREAD

**Discussion by shackra** [react-hook-form/react-hook-form#5763] [Q&A] [answered]
Hi!

I noticed that with v7.9.0 I cannot name my nested field arrays within field arrays as `${thing}.${fieldarrayID}.${nestedFieldArrayID}.value` due to type errors.

How should I work with nested field arrays from now on?

EDIT: Quick example put together in CSB https://codesandbox.io/s/react-hook-form-naming-problem-with-nested-field-arrays-9qjfo?file=/src/App.tsx:722-787

**Comment by bluebill1049** [comment, marked as answer]
can you share a CSB?

**Reply by shackra** [reply]
> can you share a CSB?

Thanks for the reply! Yes, I put together a CodeSandBox example: https://codesandbox.io/s/react-hook-form-naming-problem-with-nested-field-arrays-9qjfo?file=/src/App.tsx:722-787

please notice the error 

```
Argument of type 'string' is not assignable to parameter of type '`${string}` | `${string}.${string}` | `${string}.${number}`'
```

also see `subFieldArray.tsx`, [it displays the same error](https://codesandbox.io/s/react-hook-form-naming-problem-with-nested-field-arrays-9qjfo?file=/src/subFieldArray.tsx:414-490)

How should I work with nested field arrays from now on?

**Reply by bluebill1049** [reply]
<img width="900" alt="Screen Shot 2021-06-29 at 10 09 22 am" src="https://user-images.githubusercontent.com/10513364/123718278-160ae580-d8c2-11eb-9037-9c1439209a43.png">

is this what you after?

**Reply by shackra** [reply]
> <img alt="Screen Shot 2021-06-29 at 10 09 22 am" width="900" src="https://user-images.githubusercontent.com/10513364/123718278-160ae580-d8c2-11eb-9037-9c1439209a43.png">
> 
> is this what you after?

I'm honestly not sure. How am I supposed to know what's the value of an element in an array beforehand so I can cast it against its name?

**Reply by shackra** [reply]
@bluebill1049 maybe forking and modifying my CodeSandBox will help, illustrating that which I cannot wrap my head around from RHF documentation?

**Reply by bluebill1049** [reply]
<input {...register(`lvl1.${field.id}.item` as `lvl1.0.item`)} />

**Reply by shackra** [reply]
> <input {...register(`lvl1.${field.id}.item` as `lvl1.0.item`)} />

so I did a small experiment in [this fork](https://codesandbox.io/s/react-hook-form-naming-problem-with-nested-field-arrays-forked-2ntev?fontsize=14&hidenavigation=1&theme=dark) of my sandbox, and now I get it: casting with a literal string does not change much where RFH stores the values of the field array, I could use `"lvl1.0.lvl2.0.stat"`, `"lvl1.99.lvl2.1337.stat"` or even `"lvl1.e.lvl2.x.stat"` for the casting if I wish

also, instead of using the ID of the field (via `field.id`), I need to use the index value provided by the `.map` method when naming my inputs, otherwise the values will not be stored at all by RHF in the array field.

Now all makes more sense. Gracias!
